.github/workflows/image-push.yml

```diff
@@ -23,33 +23,26 @@ jobs:
         echo "$(make -s log | grep TAG)" >> "$GITHUB_ENV"
         echo "$(make -s log | grep GOLANG_VERSION)" >> "$GITHUB_ENV"
 
-    - name: Set up QEMU
-      uses: docker/setup-qemu-action@v3
-
-    - name: Set up Docker Buildx
-      uses: docker/setup-buildx-action@v3
-
     - name: "Read secrets"
       uses: rancher-eio/read-vault-secrets@main
       with:
         secrets: |
           secret/data/github/repo/${{ github.repository }}/dockerhub/${{ github.repository_owner }}/credentials username | DOCKER_USERNAME ;
-          secret/data/github/repo/${{ github.repository }}/dockerhub/${{ github.repository_owner }}/credentials password | DOCKER_PASSWORD
-
-    - name: Login to Container Registry
-      uses: docker/login-action@v3
-      with:
-        username: ${{ env.DOCKER_USERNAME }}
-        password: ${{ env.DOCKER_PASSWORD }}
+          secret/data/github/repo/${{ github.repository }}/dockerhub/${{ github.repository_owner }}/credentials password | DOCKER_PASSWORD ;
+          secret/data/github/repo/${{ github.repository }}/rancher-prime-registry/credentials registry | PRIME_REGISTRY ;
+          secret/data/github/repo/${{ github.repository }}/rancher-prime-registry/credentials username | PRIME_REGISTRY_USERNAME ;
+          secret/data/github/repo/${{ github.repository }}/rancher-prime-registry/credentials password | PRIME_REGISTRY_PASSWORD
 
-    - name: Build container image
-      uses: docker/build-push-action@v5
+    - name: Build and push image
+      uses: rancher/ecm-distro-tools/actions/publish-image@master
       with:
-        context: .
-        push: true
-        tags: rancher/hardened-kubernetes:${{ github.event.release.tag_name }}
-        file: Dockerfile
-        platforms: linux/amd64,linux/arm64
-        build-args: |
-          TAG=${{ env.TAG }}
-          GO_IMAGE=rancher/hardened-build-base:${{ env.GOLANG_VERSION }}
+        image: hardened-kubernetes
+        tag: ${{ github.event.release.tag_name }}
+        public-repo: rancher
+        public-username: ${{ env.DOCKER_USERNAME }}
```

Comment:
I think we should handle sensitive data as secrets, env variables can get exposed:

```(yaml)
prime-password: ${{ secrets.PRIME_REGISTRY_PASSWORD }}
```

Reply:
We're getting those values from vault, the `"Read secrets"` step will retrieve them from vault and set it as `envs`. The Github CI will mask those, so no secrets leak.

Reply:
This data is only available through vault, and not with github secrets. This use of read-vault-secrets is pretty much standardized.

Reply:
The EIO vault doesn't use gh secrets, just envs, but it still applies the same obfuscation in logs